apps/conv_layer/CMakeLists.txt

```diff
@@ -1,19 +1,15 @@
-# Generator
-halide_add_generator(conv_layer.generator
-                     SRCS conv_layer_generator.cpp)
-halide_add_aot_library(conv_layer
-                       GENERATOR_TARGET conv_layer.generator
-                       GENERATOR_ARGS target=host auto_schedule=false
-                       GENERATED_FUNCTION conv_layer)
-halide_add_aot_library(conv_layer_auto_schedule
-                       GENERATOR_NAME conv_layer
-                       GENERATOR_TARGET conv_layer.generator
-                       GENERATOR_ARGS target=host auto_schedule=true
-                       GENERATED_FUNCTION conv_layer_auto_schedule)
-
-# Final executable
 add_executable(conv_layer_process process.cpp)
-halide_add_aot_library_dependency(conv_layer_process conv_layer)
-halide_add_aot_library_dependency(conv_layer_process conv_layer_auto_schedule)
 halide_use_image_io(conv_layer_process)
-target_compile_options(conv_layer_process PRIVATE "-std=c++11")
+
+halide_generator(conv_layer.generator SRCS conv_layer_generator.cpp)
+foreach(AUTO_SCHEDULE false true)
```

Comment:
Is it worth encapsulating this in a CMake macro thingy in the support dir?

Reply:
My first inclination is "no, not yet" -- this isn't really complex enough to need a helper IMHO, and I think adding the helper might just add layer of obfuscation to the new rules (plus it would be something for people to cargo-cult).

Adding well-defined patterns for auto-schedule is something we definitely want to do, but I'm inclined to avoid it in the initial rule set.